docs/Conformance-Guide-draft.md

```diff
@@ -0,0 +1,103 @@
+
+   The package.json file must be provided to verify this criteria.
+
+   ```json
+   "extensionDependencies": [
+    "Zowe.vscode-extension-for-zowe"
+   ],
+   ```
+
+1. (Required if VS Code) Zowe Extender access: **Zowe Explorer VS Code extensions must access the shared Zowe Explorer profiles cache only via the `ZoweExplorerApi.IApiRegisterClient.getExplorerExtenderApi()` API as documented in the Zowe Explorer extensibility documentation.**
+
+   API calls to the Explorer Extender API will be logged by Zowe Explorer and be verified by inspecting Zowe Explorer log files.
+
+1. (Required if VS Code) Added Profile Type initialization: **If the extender depends on a new Zowe CLI profile type other than the Zowe Explorer default `zosmf` then an extender needs to call `ZoweExplorerApi.IApiRegisterClient.getExplorerExtenderApi().initialize(profileTypeName)`**
+
+   **(TBD)** as its first API call to ensure that the profile type can be supported. This call ensures that the profiles of this type can be successfully managed even without the Zowe CLI plugin being installed locally for pure VS Code users as well as that the profiles can successfully be accessed in the secure credentials store. This registration call will be logged by Zowe Explorer and be verified by inspecting Zowe Explorer log files.
+
+## Data Provider extension conformance criteria (2)
+
+A data provider Zowe Explorer extension provides an alternative protocol for Zowe Explorer to interact with z/OS. The default protocol Zowe Explorer uses is the z/OSMF REST APIs and data provider adds support for another API. For example, the Zowe Explorer extension for zFTP, which is maintained by the Zowe Explorer squad is an example for a Zowe Explorer data provider extension that uses FTP instead of z/OSMF for all of its USS and MVS interactions. To achieve such an extension it uses a Zowe CLI Plugin for FTP that implemented the core interactions and provided them as an SDK andd
+
+1. (Required) New Zowe CLI profile type: **A new data provider requires a new Zowe CLI profile type as it is the foundation for registering the APIs that add the new protocol as documented in the extension guideline. When new APIs get registered via the `ZoweExplorerApi.IApiRegisterClient.registerMvsApi()` call the APIs passed as a parameter needs to use a unique profile type name.**
+
+   Such registration events will be logged by Zowe Explorer and can be used to verify to successful implementation of the requirement in the Zowe Explorer logs.
+
+1. (Best Practice) Matching Zowe CLI Plugin: **Provide a Zowe CLI Plugin for the data provider's profile type that implements the core capabilities requires for the new protocol that users can then also use to interact with the protocol outside of the Zowe Explorer extension using Zowe CLI commands.**
```

Comment:
If the extension has a Zowe CLI counterpart, it must use the functionalities implemented in the CLI. ??

I wanted to say that if the extension has a CLI, then the functionalities will be shared with VSCode (e.g FTP)

I am not sure if this should just under a data provider or it should be under general. I could have a Zowe CLI Plugin but not a data provider. I could just extend the menu and have a matching plugin and I think sharing functionalities with CLI (in general) should be a requirement

Reply:
Made this a best practice as one could write a data provider without CLI.